src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,62 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String COMA_REGEX = ",";
+    private static final String STRING_BUY = "buy";
+    private static final String STRING_SUPPLY = "supply";
+    private static final String STRING_RESULT = "result";
+    private static final int BUY_SUPPLY_INDEX = 0;
+    private static final int VALUE_INDEX = 1;
+    private int supplySum;
+    private int buySum;
 
     public void getStatistic(String fromFileName, String toFileName) {
+        readFromFile(fromFileName);
+        String report = createReport();
+        writeIntoNewFile(report, toFileName);
+    }
+
+    private void readFromFile(String fromFileName) {
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(fromFileName));
```

Comment:
Reader should be closed in block 'finally'. But it's better to use "try with resources" for autoclosing
```suggestion
            try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFile))) {
```